x-pack/plugin/repository-encrypted/src/internalClusterTest/java/org/elasticsearch/repositories/encrypted/EncryptedAzureBlobStoreRepositoryIntegTests.java

```diff
@@ -100,4 +103,13 @@ protected long blobLengthFromContentLength(long contentLength) {
         return getEncryptedBlobByteLength(contentLength);
     }
 
+    @Override
+    protected BlobStore newBlobStore(String repository) {
+        BlobStore blobStore = super.newBlobStore(repository);
+        final EncryptedRepository encryptedRepository = (EncryptedRepository) internalCluster().getCurrentMasterNodeInstance(
+            RepositoriesService.class
+        ).repository(repository);
+        PlainActionFuture.get(encryptedRepository::publishPasswordsHashes);
```

Comment:
The tests that directly access the blobstore and get an instance via this method usually write a blob.
But writing a blob triggers writing an encrypted DEK, which requires verifying the password hashes, but which
are not otherwise published in these artificial cases; in production code, blob writes are always preceded by publishing hashes.